pubmed_dl/main.py

```diff
@@ -1,38 +1,55 @@
-import json
-from pubmed_dl.ncbi import get_list_pmid, uids_to_docs
-import logging
-import os
-from pathlib import Path
-from dotenv import load_dotenv
-from pydantic import BaseSettings
-import time
-
-
-dot_env_filepath = Path(__file__).absolute().parent.parent / ".env"
-load_dotenv(dot_env_filepath)
-
-
-class Settings(BaseSettings):
-    loglevel: str = os.getenv("LOGLEVEL", "INFO")
-
-
-settings = Settings()
-logging.basicConfig(level=settings.loglevel)
-
-
-def main():
-    start_time = time.time()
-    start = input("Enter start date (YYYY/MM/DD): ")
-    end = input("Enter end date (YYYY/MM/DD): ")
-    val = get_list_pmid(start, end)
-    with open("test.json", "w") as f:
-        for batch in uids_to_docs(val):
-            for doc in batch:
-                f.write(f"{json.dumps(doc)}\n")
-    logging.info(f"Done writing data from {start} to {end} onto file named: test.json")
-    duration = time.time() - start_time
-    logging.info(f"Total run time for {len(val)} docs: {duration}s")
-
-
-if __name__ == "__main__":
-    main()
+import json
+from pubmed_dl.ncbi import get_list_pmid, uids_to_docs
+import logging
+import os
+from pathlib import Path
+from dotenv import load_dotenv
+from pydantic import BaseSettings
+import time
+import typer
+from typing import Optional
+
+
+dot_env_filepath = Path(__file__).absolute().parent.parent / ".env"
+load_dotenv(dot_env_filepath)
+
+
+class Settings(BaseSettings):
+    loglevel: str = os.getenv("LOGLEVEL", "INFO")
+
+
+settings = Settings()
+logging.basicConfig(level=settings.loglevel)
+
+
+def main(start: str, end: str, request_file: Optional[str] = typer.Argument(None)):
```

Comment:
I wouldn't make the output file path a default unless there's a good argument for it. Output file paths are almost always required arguments for CLIs

Reply:
I made it optional in the sense that if it's not provided then it writes to ```test.json```. However, if it is provided, then it goes there. As of right now it always writes to a file, it's either to ```test.json```(if nothing is entered) or it is what they request. But I can change it to required instead of optional, it's your call.

I'm trying to figure out how the directory thing works now, the one you sent above.

Reply:
I would make it required unless there's a good argument for "test.json" being the default.

Reply:
okay let me try that. Also I'll make the argument required

Reply:
Done